src/3box.js

```diff
@@ -96,12 +96,16 @@ class Box {
     this.replicator.rootstore.setIdentity(await this._3id.getOdbId())
     this.syncDone = this.replicator.syncDone
 
+    if (this._3id.idWallet) {
+      this._3id.idWallet.events.on('new-auth-method', authData => {
+        this._writeRootstoreEntry(Replicator.entryTypes.AUTH_DATA, authData)
+      })
+    }
+
     this.public = new PublicStore(this._3id.muportFingerprint + '.public', this._linkProfile.bind(this), this.replicator, this._3id)
     this.private = new PrivateStore(this._3id.muportFingerprint + '.private', this.replicator, this._3id)
-    await Promise.all([
-      this.public._load(),
-      this.private._load()
-    ])
+    await this.public._load()
+    await this.private._load()
```

Comment:
prior implementation would have been faster in most env, unless you need public load to resolve before private load

Reply:
I was seeing some weird concurrency issue here when running the integration test. I don't think this would be much slower at all tbh!